GamepadButtonAndAxisEvents/explainer.md

```diff
@@ -0,0 +1,216 @@
+  readonly attribute boolean connected;
+  readonly attribute DOMHighResTimeStamp timestamp;
+  readonly attribute GamepadMappingType mapping;
+  readonly attribute FrozenArray<double> axes;
+  readonly attribute FrozenArray<GamepadButton> buttons;
+};
+```
+### RawGamepadInputChangeEvent interface IDL, used for rawgamepadinputchange.
+```
+[Exposed=Window, SecureContent]
+interface RawGamepadInputChangeEvent : Event {
+  constructor(DOMString type, optional RawGamepadInputChangeEventInit eventInitDict = {});
+
+  readonly attribute Gamepad gamepadSnapshot;
+  readonly attribute FrozenArray<unsigned long> axesChanged;
+  readonly attribute FrozenArray<unsigned long> buttonsValueChanged;
+  readonly attribute FrozenArray<unsigned long> buttonsPressed;
+  readonly attribute FrozenArray<unsigned long> buttonsTouched;
+};
+```
+##  Developer code sample
+
+```JS
+// Listen for when a gamepad is connected
+window.ongamepadconnected = (connectEvent) => {
+
+  const connectedGamepads = navigator.getGamepads();
+
+  const gamepad = connectedGamepads[connectEvent.gamepad.index];
+
+  console.log(`Gamepad connected: ${gamepad.id} (index: ${gamepad.index})`);
+
+  // Listen for input changes on this gamepad.
+  gamepad.onrawgamepadinputchange = (changeEvent) => {
+    const snapshot = changeEvent.gamepadSnapshot;
+
+    for (const axisIndex of changeEvent.axesChanged) {
+      const axisValue = snapshot.axes[axisIndex];
+      console.log(`Axis ${axisIndex} on gamepad ${snapshot.index} changed to ${axisValue}`);
+    }
+
+    // Analog buttons (ex: triggers)
+    for (let buttonIndex of changeEvent.buttonsValueChanged) {
+      const buttonValue = changeEvent.gamepadSnapshot.buttons[buttonIndex].value;
+      console.log('button ' + buttonIndex +
+                  ' on gamepad ' + changeEvent.gamepadSnapshot.index +
+                  ' changed to value ' + buttonValue);
+    }
+
+    // Binary buttons pressed
+    for (let buttonIndex of changeEvent.buttonsPressed) {
+      const buttonPressed = changeEvent.gamepadSnapshot.buttons[buttonIndex].pressed;
+      console.log('button ' + buttonIndex +
+                  ' on gamepad ' + changeEvent.gamepadSnapshot.index +
+                  ' changed pressed to ' + buttonPressed);
+    }
+
+    // Buttons touched
+    for (let buttonIndex of changeEvent.buttonsTouched) {
+      const buttonTouched = changeEvent.gamepadSnapshot.buttons[buttonIndex].touched;
+      console.log('button ' + buttonIndex +
+                  ' on gamepad ' + changeEvent.gamepadSnapshot.index +
+                  ' changed touched to ' + buttonTouched);
+    }
+  };
+};
+
+```
+
+### Dependencies on non-stable features
+
+None identified at this stage. The proposal builds upon the existing Gamepad API.
+
+## Alternatives considered
+gamepadchange event: Similar to rawgamepadinputchange event but instead the getCoalescedEvents() method is used to return a sequence of events that have been coalesced (combined) together.
```

Comment:
Maybe add more context about it? Like the proposed IDL and a link to where this was proposed and discussed?